sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -713,3 +713,9 @@ export function getRandomIntegerInclusive(min: number, max: number): number {
   const offset = Math.floor(Math.random() * (max - min + 1));
   return offset + min;
 }
+
+export function modifySpanOptions(options: OperationOptions): void {
+  if (options.tracingOptions) {
+    options.tracingOptions.spanOptions = (options.tracingOptions as any)?.spanOptions;
+  }
+}
```

Comment:
I am confused by this helper. It desugars in JS to:

```js
if (options.tracingOptions) {
   options.tracingOptions.spanOptions = options.tracingOptions.spanOptions;
}
```

This seems like it doesn't actually do anything. Can you elaborate about why it was needed?

Reply:
We used to call `operationOptionsToRequestOptionsBase(updatedOptions)` in these methods. We do not have equivalent method in core-client/core-rest-pipeline. So, when I checked the method:

```
export function operationOptionsToRequestOptionsBase<T extends OperationOptions>(
  opts: T
): RequestOptionsBase {
  const { requestOptions, tracingOptions, ...additionalOptions } = opts;

  let result: RequestOptionsBase = additionalOptions;

  if (requestOptions) {
    result = { ...result, ...requestOptions };
  }

  if (tracingOptions) {
    result.tracingContext = tracingOptions.tracingContext;
    // By passing spanOptions if they exist at runtime, we're backwards compatible with @azure/core-tracing@preview.13 and earlier.
    result.spanOptions = (tracingOptions as any)?.spanOptions;
  }

  return result;
}
```

The only thing I see in that menthod was the assignment of spanOptions. That was the reason I added this line.

Reply:
Right, but that was because we were creating a new shape (RequestOptionsBase) that had the key in a different place, e.g.

```ts
let newOptions: RequestOptionsBase = {};

if(oldOptions.tracingOptions) {
  newOptions.spanOptions = oldOptions.tracingOptions.spanOptions
}
```

But now we're not creating a new options object, nor are we changing the location from `.tracingOptions.spanOptions` to `.spanOptions`, so it's not needed.

Reply:
Done